TF_Tooling_Web/src/Cursors/tests/SvgCursor.test.ts

```diff
@@ -0,0 +1,118 @@
+import TouchFree from '../../TouchFree';
+import { HandChirality, HandType, InputType, InteractionType, TouchFreeInputAction } from '../../TouchFreeToolingTypes';
+import { SVGCursor } from '../SvgCursor';
+
+TouchFree.Init();
+const svgCursor = new SVGCursor();
+const cursor = document.getElementById('svg-cursor');
+const cursorRing = document.getElementById('svg-cursor-ring');
+const cursorDot = document.getElementById('svg-cursor-dot');
+
+const mockTfInputAction = (input?: Partial<TouchFreeInputAction>) =>
+    TouchFree.DispatchEvent(
+        'TransmitInputAction',
+        new TouchFreeInputAction(
+            input?.Timestamp ?? Date.now(),
+            input?.InteractionType ?? InteractionType.PUSH,
+            input?.HandType ?? HandType.PRIMARY,
+            input?.Chirality ?? HandChirality.RIGHT,
+            input?.InputType ?? InputType.MOVE,
+            input?.CursorPosition ?? [0, 0],
+            input?.DistanceFromScreen ?? 5,
+            input?.ProgressToClick ?? 0
+        )
+    );
+
+describe('SVG Cursor', () => {
+    beforeAll(() => {
+        // Set cursor to known state before each test
+        mockTfInputAction();
+        mockTfInputAction({ InputType: InputType.UP });
+        svgCursor.EnableCursor();
+        svgCursor.ShowCursor();
+    });
+
+    test('Creates a cursor in the document body', () => {
+        expect(cursor).toBeTruthy();
+        expect(cursorDot).toBeTruthy();
+        expect(cursorRing).toBeTruthy();
+    });
+
+    test('Update cursor position when MOVE action received', () => {
+        mockTfInputAction({ InputType: InputType.MOVE, CursorPosition: [100, 100] });
+        expect(cursorDot?.getAttribute('cx')).toBe('100');
+        expect(cursorRing?.getAttribute('cy')).toBe('100');
+    });
+
+    test('Cursor ring should grow with ProgressToClick', () => {
+        mockTfInputAction({ InputType: InputType.MOVE, ProgressToClick: 0.1 });
+
+        expect(cursorRing?.getAttribute('r')).toBe('29');
```

Comment:
:shrug: Should we check attribute before to check it's grown by the percentage we expect :shrug:

Reply:
Done!